> Thanks @tejlmand . From #65995 I see you have a lot of experience with system (mis)configuration so can please give just one example of stderr noise emitted by some `python` executable when trying to run `sys.version_info[:2]`?

Sure, a broken python 3.8 installation in path with current code can result in:
```
-- Application: /projects/github/ncs/zephyr/samples/hello_world
-- CMake version: 3.26.4
-- Found Python3: /usr/bin/python3.11 (found suitable version "3.11.4", minimum required is "3.8") found components: Interpreter 
```
which looks very clean.


But with this PR, then the following is seen instead:
```
-- Application: /projects/github/ncs/zephyr/samples/hello_world
-- CMake version: 3.26.4
Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
Python path configuration:
  PYTHONHOME = (not set)
  PYTHONPATH = (not set)
  program name = '/usr/bin/python3'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = '/usr/bin/python3'
  sys.base_prefix = '/usr'
  sys.base_exec_prefix = '/usr'
  sys.executable = '/usr/bin/python3'
  sys.prefix = '/usr'
  sys.exec_prefix = '/usr'
  sys.path = [
    '/usr/lib/python38.zip',
    '/usr/lib/python3.8',
    '/usr/lib/lib-dynload',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007f17cefca740 (most recent call first):
<no Python frame>
-- Found Python3: /usr/bin/python3.11 (found suitable version "3.11.4", minimum required is "3.8") found components: Interpreter 
```

which is noise irrelevant for the user in this case, cause the error is printed by Python 3.8, but as you see in the last line we're going to use Python 3.11, in which case the error message will just be confusing.

Sure the user has a broken Python in his system, but that's not Zephyr build system responsibility to take care of that and one must expect that he discovers such a situation running other tools.

Now breaking all my Python installations results in the standard CMake find python error message, like this:
```
-- CMake version: 3.26.4
CMake Error at /opt/cmake-3.26.4-linux-x86_64/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Python3_EXECUTABLE Interpreter) (Required
  is at least version "3.8")

      Reason given by package: 
          Interpreter: Wrong version for the interpreter "/bin/python"

Call Stack (most recent call first):
  /opt/cmake-3.26.4-linux-x86_64/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /opt/cmake-3.26.4-linux-x86_64/share/cmake-3.26/Modules/FindPython/Support.cmake:3766 (find_package_handle_standard_args)
  /opt/cmake-3.26.4-linux-x86_64/share/cmake-3.26/Modules/FindPython3.cmake:551 (include)
  /projects/github/ncs/zephyr/cmake/modules/python.cmake:44 (find_package)
  /projects/github/ncs/zephyr/cmake/modules/zephyr_default.cmake:129 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:117 (include_boilerplate)
  CMakeLists.txt:5 (find_package)
```
where you clearly see this line in the appearing as reference in the stack trace:
https://github.com/zephyrproject-rtos/zephyr/blob/a25e211ad232e74269fb1f9058dbda3eeb3b86c1/cmake/modules/python.cmake#L44

also remember that our preference is to use the CMake provided mechanism for finding Python and only due to shortcomings in CMake's own implementation we have to make a pre-search for Python in the Zephyr build system.

--- a/cmake/modules/python.cmake
+++ b/cmake/modules/python.cmake
@@ -31,7 +31,6 @@ if(NOT Python3_EXECUTABLE)
                                  "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"
                          RESULT_VARIABLE result
                          OUTPUT_VARIABLE version
-                         ERROR_QUIET
                          OUTPUT_STRIP_TRAILING_WHITESPACE)
 
        if(version VERSION_LESS PYTHON_MINIMUM_REQUIRED)

